### ISSUE TITLE
How to add non-heading type sections in the tocbot output

### ISSUE THREAD

**Issue by kapooramit** [tscanlin/tocbot#270]
Hi @tscanlin 

We have long API class pages that list methods. Each method (in Sphinx HTML) is shown as '<dl>' type with a class. So there is no heading style. I there a way that tocbot can be asked to use a custom class to in addition to heading2 to create the tocbot.
````
<dl class="py method">
  <dt class="sig sig-object py" id="product.metaset.get_fullname">
  ..........
  </dt>
  <dd>
  .....
  
  </dd>
</dl>
````

If not, I will have to use javascript to read and add invisible headings and have tocbot use them instead. But the former would be native and efficient approach if it is possible. 

Is it possible? If yes, would you please share how?

**Comment by tscanlin** [comment]
Yeah that should be possible by setting the heading selector to target those elements. Although I don't know how nesting will work then because those tags won't have a level like the headings have.. so this may not be really easy to do.
`headingSelector: 'dt.sig',`

**Comment by kapooramit** [comment]
Thanks @tscanlin . This will work for me. I will need to update the page DOM to fix the hierarchy but it will work. Thanks a ton!

**Comment by kapooramit** [comment]
@tscanlin : It worked just fine. The only issue would be what you said - hierarchy. I am unable to find anything on web that allows me to find the last heading level before this class type (dt.sig) so that I can add appropriate heading level...

**Comment by kapooramit** [comment]
I am using JavaScript to add id and class to each (.py.method > .sig.sig-object > .sig-name.descname > .pre) and then I call tocbot. The links work but the highlighting is not working. 

I have shared the link to the live page in email  to you with subject "Sphinx tocbot issue"

I am not sure why tabindex=-1 getting added to target spans but even if remove them, the issue exists. 

Please help!

**Comment by kapooramit** [comment]
Ok. Just going to doctor sometimes solves the issues without medication 😊. And that is what happened today. 

I changed the logic to add a heading instead of updating the existing span but it didn’t work. Then, I added the heading before the first selector /span where it is actually a proper heading and it worked! Is it that the heading must be at its own level and not under something else? 

I had to set its visibility to hidden and height and width to 0px so that it is not shown and doesn’t take any space.

New working HTML:
```
<h3 id="get_child_objects" class="apifunction">get_child_objects</h3>
<dl class="py method">
<dt class="sig sig-object py" id="get_child_objects">
<span class="sig-name descname">
<span class="pre">get_child_objects</span>
....
</dt>

```
Thank you!

**Comment by tscanlin** [comment]
Yeah, heading have to be at their own level and can’t be nested, that is correct. Glad you were able to make progress on this!
